File: ios/Classes/ConstraintMapper.swift
import Foundation

/// Maps TaskConstraints to iOS BGTask request properties.
struct ConstraintMapper {
  static func requiresNetwork(_ map: [String: Any]?) -> Bool {
    guard let network = map?["network"] as? String else { return false }
    return network == "connected" || network == "unmetered"
  }

  static func requiresCharging(_ map: [String: Any]?) -> Bool {
    (map?["requiresCharging"] as? Bool) == true
  }

  static func warnIgnoredConstraints(_ map: [String: Any]?) {
    if (map?["deviceIdle"] as? Bool) == true {
      print("[TaskFlow] WARNING: deviceIdle constraint is not supported on iOS and will be ignored.")
    }
  }
}

File: android/src/main/kotlin/dev/taskflow/ConstraintMapper.kt
package dev.taskflow

import androidx.work.Constraints
import androidx.work.NetworkType

/** Maps TaskConstraints to Android WorkManager Constraints. */
object ConstraintMapper {
  fun fromMap(map: Map<String, Any?>?): Constraints? {
    if (map == null) return null

    val builder = Constraints.Builder()

    when (map["network"] as? String) {
      "connected" -> builder.setRequiredNetworkType(NetworkType.CONNECTED)
      "unmetered" -> builder.setRequiredNetworkType(NetworkType.UNMETERED)
    }

    if (map["batteryNotLow"] as? Boolean == true) {
      builder.setRequiresBatteryNotLow(true)
    }
    if (map["requiresCharging"] as? Boolean == true) {
      builder.setRequiresCharging(true)
    }
    if (map["deviceIdle"] as? Boolean == true) {
      builder.setRequiresDeviceIdle(true)
    }

    return builder.build()
  }
}
